R/evaluate-promise.R

```diff
@@ -39,9 +39,12 @@ evaluate_promise <- function(code, print = FALSE) {
   temp <- file()
   on.exit(close(temp))
 
-  result <- with_sink(temp,
-    withCallingHandlers(
-      withVisible(code), warning = wHandler, message = mHandler
+  with_reporter(
```

Comment:
Why can't you handle multiple warnings? Oh because the reporter also needs to print them?